File: TLBB1.5/tabletools/CodeTable/CSharp/Table_Popup.cs
//This code create by CodeEngine mrd.cyou.com ,don't modify
using System;
 using System.Collections.Generic;
 using System.Collections;

namespace GCGame.Table{

[Serializable]
 public class Tab_Popup : ITableOperate
{ private const string TAB_FILE_DATA = "popup.txt";
 public enum _ID
 {
 INVLAID_INDEX=-1,
ID_TITLE,
ID_CONTENT,
ID_TYPE,
MAX_RECORD
 }
 public string GetInstanceFile(){return TAB_FILE_DATA; }

private string m_Content;
 public string Content { get{ return m_Content;}}

private string m_Title;
 public string Title { get{ return m_Title;}}

private string m_Type;
 public string Type { get{ return m_Type;}}

public bool LoadTable(Hashtable _tab)
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),SerializableTable,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }
 public void SerializableTable(ArrayList valuesList,string skey,Hashtable _hash)
 {
 if (string.IsNullOrEmpty(skey))
 {
 throw TableException.ErrorReader("Read File{0} as key is Empty Fail!!!", GetInstanceFile());
 }

 if ((int)_ID.MAX_RECORD!=valuesList.Count)
 {
 throw TableException.ErrorReader("Load {0} error as CodeSize:{1} not Equal DataSize:{2}", GetInstanceFile(),_ID.MAX_RECORD,valuesList.Count);
 }
 Int32 nKey = Convert.ToInt32(skey);
 Tab_Popup _values = new Tab_Popup();
 _values.m_Content =  valuesList[(int)_ID.ID_CONTENT] as string;
_values.m_Title =  valuesList[(int)_ID.ID_TITLE] as string;
_values.m_Type =  valuesList[(int)_ID.ID_TYPE] as string;

 _hash.Add(nKey,_values); }


}
}

File: TLBB1.5/tabletools/CodeTable/Java/xjgame/table/Table_Popup.java
//This code create by CodeEngine mrd.cyou.com ,don't modify
package xjgame.table;
import java.util.ArrayList;
 import java.util.HashMap;

public class Table_Popup implements ITableOperate {
 public final static String TAB_FILE_DATA = "/popup.txt";
 public static final int ID_TITLE = 0 ;
public static final int ID_CONTENT = 1 ;
public static final int ID_TYPE = 2 ;
public static final int MAX_RECORD = 3 ;

 public String GetInstanceFile(){return TAB_FILE_DATA; }

private String m_Content;
 public String GetContent() { return m_Content;}

private String m_Title;
 public String GetTitle() { return m_Title;}

private String m_Type;
 public String GetType() { return m_Type;}

public boolean LoadTable(HashMap<Integer, Table_Popup> _tab) throws TableException
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),this,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }

 @Override
 public void SerializableTable(ArrayList<String> valuesList,String skey,Object obj) throws TableException {
 @SuppressWarnings("unchecked")
 HashMap<Integer, Table_Popup> _hash = (HashMap<Integer, Table_Popup> )obj;
 if (skey.isEmpty())
 {
 throw TableException.ErrorReader("Read File"+GetInstanceFile()+" as key is Empty Fail!!!");
 }

 if (MAX_RECORD!=valuesList.size())
 {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as CodeSize:"+MAX_RECORD+" not Equal DataSize:"+valuesList.size());
 }
 try { 
 Integer nKey = Integer.parseInt(skey);
 Table_Popup _values = new Table_Popup();
 _values.m_Content =  (String)valuesList.get(ID_CONTENT);
_values.m_Title =  (String)valuesList.get(ID_TITLE);
_values.m_Type =  (String)valuesList.get(ID_TYPE);

 _hash.put(nKey,_values);
 } catch (NumberFormatException e) {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as :" + e.getMessage());
 }
 }


}

